File: services/python-agent/app/api/v1/endpoints/health.py
"""
Endpoints de health check e readiness.

Usados pelo gateway Go e orquestrador para verificar status do servico.
"""

from __future__ import annotations

from fastapi import APIRouter

router = APIRouter()


@router.get("/health")
def health_check() -> dict[str, str]:
    """Health check basico — indica que o servico esta no ar."""
    return {"status": "ok"}


@router.get("/ready")
def readiness_check() -> dict[str, str | dict[str, str]]:
    """Readiness check — verifica dependencias (mock por enquanto)."""
    return {
        "status": "ready",
        "checks": {
            "database": "ok",
        },
    }

File: services/go-gateway/internal/api/v1/handlers/health.go
package handlers

import (
	"net/http"

	"github.com/gin-gonic/gin"
)

// Health retorna status basico do servico.
// GET /api/v1/health
func Health(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{"status": "ok"})
}

// Ready indica que o servico esta pronto para receber requisicoes.
// GET /api/v1/health/ready
func Ready(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{"status": "ready"})
}
